atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ConflictDetectionManager.java

```
@@ -15,56 +15,133 @@
+                    }
+                });
+    }
+
+    @VisibleForTesting
+    public static ConflictDetectionManager createWithStaticConflictDetection(
+            Map<TableReference, ConflictHandler> staticMap) {
+        CacheLoader<TableReference, ConflictHandler> loader = new CacheLoader<TableReference, ConflictHandler>() {
+            @Override
+            public ConflictHandler load(TableReference tableReference) throws Exception {
+                ConflictHandler conflictHandler = staticMap.get(tableReference);
+                if (conflictHandler == null) {
+                    return ConflictHandler.RETRY_ON_WRITE_WRITE; // legacy behavior / this is dumb
+                }
+                return conflictHandler;
+            }
+        };
+        return new ConflictDetectionManager(
+                loader);
     }
 
-    public boolean isEmptyOrContainsTable(TableReference tableRef) {
-        Map<TableReference, ConflictHandler> tableToConflict = cachedResult.get();
-        return tableToConflict.isEmpty() || tableToConflict.containsKey(tableRef);
+    private static ConflictHandler getConflictHandlerFromMetadata(byte[] metadata) {
+        return TableMetadata.BYTES_HYDRATOR
+                .hydrateFromBytes(metadata).getConflictHandler();
     }
 
-    public Map<TableReference, ConflictHandler> get() {
-        return cachedResult.get();
+    private void warmCacheWith(Map<TableReference, ConflictHandler> preload) {
+        cache.putAll(preload);
     }
 
-    public void recompute() {
-        cachedResult.recompute();
+    @VisibleForTesting
+    public Map<TableReference, ConflictHandler> getCachedValues() {
+        return cache.asMap();
+    }
+
+    public ConflictHandler get(TableReference tableReference) {
+        ConflictHandler conflictHandler = cache.getUnchecked(tableReference);
+        if (conflictHandler == null) {
+            return ConflictHandler.RETRY_ON_WRITE_WRITE; // legacy behavior / lame
```

I'm hoping to get rid of the RETRY_ON_WRITE_WRITE bullshit once I determine nothing is using this shitty silent fallback behavior